### PR TITLE
[infra] Remove custom import remapping in tests

### PR DESCRIPTION
**Pull request by justinfagnani**
Attempts to fix https://github.com/lit/lit/issues/2844 for just the lit-html package.

There's a perplexing problem with `private-ssr-support.js` though, where the identifiers it uses to pull off values from lit-html's private exports object (`_$LH`) are being minified differently than the identifiers in `lit-html.js`, leading to `private-ssr-support.js` export object having incorrect values.

### CODE REVIEW

File: packages/lit-html/package.json [hunk 1]
```diff
@@ -14,114 +14,142 @@
   "type": "module",
   "exports": {
     ".": {
+      "types": "./development/lit-html.d.ts",
```
Comment (aomarks) [inline packages/lit-html/package.json:17]: So does this take the place of having to copy types to the root of the package? Can we remove treemirror now? Or do we still need it for back-compat with older TypeScript versions.

File: packages/tests/web-test-runner.config.js [hunk 2]
```diff
@@ -6,32 +6,28 @@
...
-  resolveRemapConfig = devResolveRemapConfig;
 }
 
 const browserPresets = {
   // Default set of Playwright browsers to test when running locally.
   local: [
     'chromium', // keep browsers on separate lines
-    'firefox', // to make it easier to comment out
-    'webkit', // individual browsers
+    // 'firefox', // to make it easier to comment out
```
Comment (aomarks) [inline packages/tests/web-test-runner.config.js:29]: Uncomment

Comment (aomarks) [inline packages/tests/web-test-runner.config.js:29]: Remove debugging comment-outs

File: packages/tests/web-test-runner.config.js [hunk 3]
```diff
@@ -176,12 +172,13 @@ export default {
     '../lit-html/development/**/*_test.(js|html)',
     '../reactive-element/development/**/*_test.(js|html)',
   ],
-  nodeResolve: true,
+  nodeResolve: {
+    exportConditions: [exportCondition],
```
Comment (aomarks) [inline packages/tests/web-test-runner.config.js:176]: Awesome!

Comment (aomarks) [inline packages/tests/web-test-runner.config.js:176]: Awesome!

File: packages/tests/wtr-config.js [hunk 4]
```diff
@@ -1,57 +0,0 @@
-/**
```
Comment (aomarks) [inline packages/tests/wtr-config.js:1]: I think this file is referenced in a bunch of Wireit dependencies, can all be removed now.

Comment (aomarks) [inline packages/tests/wtr-config.js:1]: This file is referenced in a bunch of Wireit dependencies, can all be removed now.

File: packages/tests/web-test-runner.config.js [hunk 5]
```diff
@@ -212,7 +209,7 @@ export default {
     }),
   ],
   // Only actually log errors and warnings. This helps make test output less spammy.
-  filterBrowserLogs: (type) => type === 'warn' || type === 'error',
+  // filterBrowserLogs: (type) => type === 'warn' || type === 'error',
```
Comment (aomarks) [inline packages/tests/web-test-runner.config.js:212]: Uncomment

Comment (aomarks) [inline packages/tests/web-test-runner.config.js:212]: Uncomment

File: packages/lit-html/src/test/lit-html_test.ts [hunk 6]
```diff
@@ -1626,7 +1628,7 @@ suite('lit-html', () => {
       assertRender(result, '<div>A</div>');
     });
 
-    test('attribute expression', () => {
+    test.only('attribute expression', () => {
```
Comment (aomarks) [inline packages/lit-html/src/test/lit-html_test.ts:1631]: Un-only

Comment (aomarks) [inline packages/lit-html/src/test/lit-html_test.ts:1631]: Un-only

File: packages/lit-html/src/test/lit-html_test.ts [hunk 7]
```diff
@@ -862,7 +864,7 @@ suite('lit-html', () => {
     });
 
     test.skip('renders a Symbol to an attribute', () => {
-      render(html`<div foo=${Symbol('A')}></div>`, container);
+      render(html`<div foo=${Symbol('A') as any}></div>`, container);
```
Comment (aomarks) [inline packages/lit-html/src/test/lit-html_test.ts:867]: Why do we need the cast now?

Comment (aomarks) [inline packages/lit-html/src/test/lit-html_test.ts:867]: Why do we need the cast now?

File: packages/lit-html/src/private-ssr-support.ts [hunk 8]
```diff
@@ -19,6 +19,11 @@ import {
 } from './lit-html.js';
 export type {Template} from './lit-html.js';
 
+const DEV_MODE = true;
+
+console.log('DEV_MODE', DEV_MODE);
```
Comment (aomarks) [inline packages/lit-html/src/private-ssr-support.ts:24]: Remove debugging

Comment (aomarks) [inline packages/lit-html/src/private-ssr-support.ts:24]: Remove debugging

File: packages/lit-html/package.json [hunk 9]
```diff
@@ -264,7 +292,7 @@
       "output": []
     },
     "test:prod": {
-      "command": "MODE=prod node ../tests/run-web-tests.js \"development/**/*_test.(js|html)\" --config ../tests/web-test-runner.config.js",
+      "command": "MODE=prod node ../tests/run-web-tests.js \"development/**/lit-html_test.js\" --config ../tests/web-test-runner.config.js",
```
Comment (aomarks) [inline packages/lit-html/package.json:295]: Is this debugging? We want to keep the `*`.

Comment (aomarks) [inline packages/lit-html/package.json:295]: Is this debugging? We want to keep the `*`.

File: lit-next.code-workspace [hunk 10]
```diff
@@ -58,7 +58,7 @@
     }
   ],
   "settings": {
-    "typescript.tsdk": "node_modules/typescript/lib",
+    "typescript.tsdk": "lit-html/node_modules/typescript/lib",
```
Comment (aomarks) [inline lit-next.code-workspace:61]: Debugging?

Comment (aomarks) [inline lit-next.code-workspace:61]: Debugging?

**Review by aomarks — CHANGES_REQUESTED** [review]

File: packages/tests/web-test-runner.config.js (hunk 2 above)
Comment (justinfagnani) [reply packages/tests/web-test-runner.config.js:29]: Yes, I'm going to put all this back in place. I uploaded and made a draft PR to show what I had done so far and talk about the renaming problem I ran into.

File: packages/lit-html/src/test/lit-html_test.ts (hunk 7 above)
Comment (justinfagnani) [reply packages/lit-html/src/test/lit-html_test.ts:867]: Something with TS 4.7

File: packages/lit-html/src/private-ssr-support.ts (hunk 8 above)
Comment (justinfagnani) [reply packages/lit-html/src/private-ssr-support.ts:24]: I will when this is ready to go

File: lit-next.code-workspace (hunk 10 above)
Comment (justinfagnani) [reply lit-next.code-workspace:61]: This is just so I don't see errors in VS code that aren't there when building due to the difference TS versions. If we can figure out the renaming issue, this I would upgrade the rest of the repo. This wouldn't be merged as-is.

**Comment by augustjk** [comment]
Superseded by https://github.com/lit/lit/pull/3132

**Pull request closed without merging**